src/jit/hwintrinsiccodegenxarch.cpp

```diff
@@ -24,10 +24,100 @@ XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
 #include "gcinfo.h"
 #include "gcinfoencoder.h"
 
+//------------------------------------------------------------------------
+// genIsTableDrivenHWIntrinsic:
+//
+// Arguments:
+//    category - category of a HW intrinsic
+//
+// Return Value:
+//    returns true if this category can be table-driven in CodeGen
+//
+bool CodeGen::genIsTableDrivenHWIntrinsic(HWIntrinsicCategory category)
+{
+    // TODO - make more categories to the table-driven framework
+    const HWIntrinsicCategory drivableCategory    = HW_Category_SimpleSIMD;
+    const HWIntrinsicCategory nondrivableCategory = HW_Category_Special;
+    return (category & drivableCategory) != 0 && (category & HW_Category_Special) == 0;
```

Comment:
simplify: `category & drivableCategory` to `category & HW_Category_TableDriven` and define `HW_Category_TableDriven` as a mask of all categories which can be table driven. This allows to drop line 39 and 40 and move responsibility of defining table driven categories to definition of `HW_Category_TableDriven`. 

As a result it will be in the same place as definition of all underlying categories simplifying code reading.

Reply:
I think anything other than the special intrinsics (which, generally speaking, will have one or more other intrinsic nodes returned) should be table drivable, so i'm not sure if we'll need this check long term.

Reply:
Here is a temporary solution, I will modify it in subsequent PRs.